tests/integration/kubernetes/gha-run.sh

```diff
@@ -241,6 +241,66 @@ install_kata_tools_placeholder() {
 		"after CI picks up the gha yaml changes required to test that installation."
 }
 
+function update_worker_label(){
+	local workers
+	local label="$1"
+	workers="$(kubectl get nodes --no-headers | grep '\<worker\>' | awk '{ print $1 }')"
+	for nodename in $workers; do
+		if ! kubectl get "node/${nodename}" -o jsonpath='{.metadata.labels}' | \
+			grep "$label" >/dev/null; then
+			kubectl label node "$nodename" --overwrite "${label}"
+		fi
+	done
+}
+
+function deploy_snapshotter() {
+	echo "::group::Deploying ${SNAPSHOTTER}"
+
+	case ${SNAPSHOTTER} in
+		nydus) deploy_nydus_snapshotter ;;
+		*) >&2 echo "${SNAPSHOTTER} flavour is not supported"; exit 2 ;;
+	esac
+
+	echo "::endgroup::"
+}
+
+function deploy_nydus_snapshotter() {
+    echo "::group::deploy_nydus_snapshotter"
+
+	local nydus_snapshotter_install_dir="/tmp/nydus-snapshotter"
+	if [ -d "${nydus_snapshotter_install_dir}" ]; then
+		rm -rf "${nydus_snapshotter_install_dir}"
+	fi
+	mkdir -p "${nydus_snapshotter_install_dir}"
+	nydus_snapshotter_version=$(get_from_kata_deps "externals.nydus-snapshotter.version")
+
+	pushd "$nydus_snapshotter_install_dir"
+	# Enable guest pull feature in nydus snapshotter
+	sed -i -e "s|FS_DRIVER: .*|FS_DRIVER: \"proxy\"|g" misc/snapshotter/nydus-snapshotter.yaml
```

Comment:
IIUC here is where it should be using the `$PULL_TYPE` introduced on the previous commit. If `PULL_TYPE == guest-pull` then switch-to-use-the-proxy-fs_driver?

This point I still don't have the big picture solid on my mind. So sorry if I start talking non-sense hereon.

Is it relevant to set the `FS_DRIVER` on the ConfigMap (in misc/snapshotter/nydus-snapshotter.yaml)? Because below it disable read snapshotter config from the ConfigMap and enable system service and finally enable configuring containerd for snapshotter?

Is the table below correct? And cannot the support for all those combinations be installed on the system then a specific configuration via runtimeclass? (e.g. if runtimeclass == kata-qemu-tdx then fs_driver == fusedev) 

| storage | fs_driver  | pull type        |
| --- | --- | --- |
|fuse        | fusedev   | host-sharing |
|  ?              | proxy       | guest              |
| tarfs       |blockdev |host-sharing   |

Reply:
>IIUC here is where it should be using the $PULL_TYPE introduced on the previous commit. If PULL_TYPE == guest-pull then switch-to-use-the-proxy-fs_driver?

@wainersm Thanks for your comments. Yes, I missed it. I'll fixed it right now.

Reply:
>IIUC here is where it should be using the $PULL_TYPE introduced on the previous commit. If PULL_TYPE == guest-pull then switch-to-use-the-proxy-fs_driver?

Yes.

Reply:
>Is it relevant to set the FS_DRIVER on the ConfigMap (in misc/snapshotter/nydus-snapshotter.yaml)? Because below it disable read snapshotter config from the ConfigMap and enable system service and finally enable configuring containerd for snapshotter?

Yes, that's relevant to FS_DRIVER.

Reply:
>Is the table below correct? And cannot the support for all those combinations be installed on the system then a specific configuration via runtimeclass? (e.g. if runtimeclass == kata-qemu-tdx then fs_driver == fusedev)

Unfortunately, now the nydus snapshotter can only select one of the modes on system at the same time. Dynamic switching may cause problems https://github.com/kata-containers/kata-containers/issues/8407.